### PR TITLE
DRILL-540: Allow querying hive views in Drill

### PR DESCRIPTION
**Pull request by ihuzenko**
1. Added support for Hive views in HiveSchemaFactory
2. Extracted conversion of Hive data types from DrillHiveTable to HiveToRelDataTypeConverter 
3. Removed throwing of UnsupportedOperationException from HiveStoragePlugin 
4. Added bunch of tests for checking that querying of Hive views works fine and authorization works as described in [DRILL-540](https://issues.apache.org/jira/browse/DRILL-540)

### CODE REVIEW

**Comment by ihuzenko** [comment]
@vdiravka  please check updated table with Hive queries in the [DRILL-540](https://issues.apache.org/jira/browse/DRILL-540).

File: contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestSqlStdBasedAuthorization.java [hunk 1]
```diff
@@ -137,15 +173,10 @@ private static void generateTestData() throws Exception {
...
-  private static void createTbl(final Driver driver, final String db, final String tbl, final String tblDef,
-      final String data) throws Exception {
-    executeQuery(driver, String.format(tblDef, db, tbl));
-    executeQuery(driver, String.format("LOAD DATA LOCAL INPATH '%s' INTO TABLE %s.%s", data, db, tbl));
-  }
-
   // Irrespective of each db permissions, all dbs show up in "SHOW SCHEMAS"
   @Test
   public void showSchemas() throws Exception {
+    test("USE " + MINI_DFS_STORAGE_PLUGIN_NAME);
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestSqlStdBasedAuthorization.java]: without this option test is failing, possibly one of another tests has call to USE hive; which has negative impact on the test.

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/planner/types/HiveToRelDataTypeConverter.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+    int maxLen = TypeInfoUtils.getCharacterLengthForType(pTypeInfo);
+    RelDataType relDataType = typeFactory.createSqlType(typeName, maxLen);
+    return typeFactory.createTypeWithCharsetAndCollation(relDataType, Util.getDefaultCharset(),
+        SqlCollation.IMPLICIT);
+  }
+
+  private RelDataType getRelDataType(DecimalTypeInfo decimalTypeInfo) {
+    return typeFactory.createSqlType(SqlTypeName.DECIMAL, decimalTypeInfo.precision(), decimalTypeInfo.scale());
+  }
+}
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply contrib/storage-hive/core/src/main/java/org/apache/drill/exec/planner/types/HiveToRelDataTypeConverter.java]: done

File: contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestSqlStdBasedAuthorization.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ihuzenko) [reply contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestSqlStdBasedAuthorization.java]: Added fix with comment in HiveAuthorizationHelper .

File: contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java [hunk 3]
```diff
@@ -145,6 +148,41 @@ private static void setStorabaseBasedAuthorizationInHiveConf() {
     return hiveConfig;
   }
 
+  /**
+   * User       Groups
+   * ---------------------------
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java]: added <br/>

**Comment by ihuzenko** [comment]
@vdiravka I've addressed comments, review please.

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveAuthorizationHelper.java [hunk 4]
```diff
@@ -146,6 +147,12 @@ private void authorize(final HiveOperationType hiveOpType, final List<HivePrivil
     } catch (final HiveAccessControlException e) {
       throw e;
     } catch (final Exception e) {
+      if (toRead.size() == 1 && InfoSchemaConstants.IS_SCHEMA_NAME.equalsIgnoreCase(toRead.get(0).getObjectName())) {
```
Comment (vdiravka) [inline contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveAuthorizationHelper.java]: I think you meant `'hive.%'`, but not `'hi.%'`

Am I right, it solves the following case:
```
USE hive.some_db;
SHOW SCHEMAS LIKE 'hive.%'
```
But why it should be solved in scope of exception handling? What exception is obtained and why it is obtained?
Looks like it is a fix for consequence, but not the root cause of the issue.

File: contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java [hunk 5]
```diff
@@ -145,6 +148,41 @@ private static void setStorabaseBasedAuthorizationInHiveConf() {
     return hiveConfig;
   }
 
+  /**
+   * User       Groups
+   * <br/>
```
Comment (vdiravka) [inline contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java]: I will put a new comment, since the previous one is swallowed by your commit force-pushing.

You didn't get an idea, I suggested to add `</br>` for every line of your table (use `ctrl+q` for checking it in Intelij).
But forgot about that, the proper way is to use `<table>`, `<tr>`, `<td>` tags. Please follow it.

File: contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestSqlStdBasedAuthorization.java [hunk 6]
```diff
@@ -229,77 +297,114 @@ public void select_user1_3() throws Exception {
...
+    updateClient(usr);
+    errorMsgTestHelper(query, expectedError);
+  }
+
+  private static void createHiveView(Driver driver, String db, String viewName, String tblName) {
+    String viewFullName = db + "." + viewName;
+    String tblFullName = db + "." + tblName;
+    executeQuery(driver, String.format("CREATE OR REPLACE VIEW %s AS SELECT * FROM %s LIMIT 1", viewFullName, tblFullName));
+  }
+}
```
[the remark this answers is not in the document]
Comment (ihuzenko) [reply contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestSqlStdBasedAuthorization.java]: added lines

File: contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveAuthorizationHelper.java (hunk 4 above)
Comment (ihuzenko) [reply contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveAuthorizationHelper.java]: removed, and created separate issue https://issues.apache.org/jira/browse/DRILL-6923

File: contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java (hunk 5 above)
Comment (ihuzenko) [reply contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java]: changed to multi-line comment

Comment (vdiravka) [reply contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java]: looks good. There is no need in tags and annotations anymore

Comment (ihuzenko) [reply contrib/storage-hive/core/src/test/java/org/apache/drill/exec/impersonation/hive/TestStorageBasedHiveAuthorization.java]: fixed
